cpp/include/cudf/utilities/pinned_allocator.hpp

```diff
@@ -0,0 +1,223 @@
+/*
+ *  Copyright 2008-2022 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#pragma once
+
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <thrust/detail/config.h>
+#include <thrust/system/cuda/detail/guarded_cuda_runtime_api.h>
+#include <thrust/system/cuda/error.h>
+#include <thrust/system/system_error.h>
+
+namespace cudf::detail {
```

Comment:
This file should be in `include/cudf/detail/utilities`

Reply:
Good observation. I have moved the header and updated consumers